src/prommis/solvent_extraction/sx_model_flowsheet_dynamic.py

```diff
@@ -0,0 +1,186 @@
+""" 
+Demonstration flowsheet for dynamic state solvent extraction loading process
+using parameters and data derived from West Kentucky No. 13 coal refuse.
+
+Authors: Arkoprabho Dasgupta
+
+"""
+
+from pyomo.environ import ConcreteModel, SolverFactory, units, TransformationFactory
+
+from idaes.core import FlowDirection, FlowsheetBlock
+from idaes.core.initialization import InitializationStatus
+from idaes.core.initialization.block_triangularization import (
+    BlockTriangularizationInitializer,
+)
+
+from prommis.leaching.leach_solution_properties import LeachSolutionParameters
+from prommis.solvent_extraction.ree_og_distribution import REESolExOgParameters
+from prommis.solvent_extraction.solvent_extraction import SolventExtraction
+
+"""
+Method of building a dynamic solvent extraction model with a specified number of 
+stages and with two separate property packages for the two inlet streams.
+This is a loading operation, so no additional argument has to be specified.
+
+"""
+
+m = ConcreteModel()
+
+time_duration = 60
+
+m.fs = FlowsheetBlock(dynamic=True, time_set=[0, time_duration], time_units=units.hour)
```

Comment:
Make sure you are doing a unit consistency check on this flowsheet - I am not sure if the framework will do all the necessary unit conversion if units in the time domain do not match the units sued in the properties/unit models.

Reply:
You need to check to be sure however.

Reply:
I have rechecked the equation, both sides of the material balance equation are mol/hour, regarding volume however I saw that the units are always converted to m3, even if I fix the units as L while fixing the volume. Not that it is affecting the overall equation in anyway

Reply:
You need to have a programmatic check (i.e. a test) to ensure this. The diagnostics toolbox can do this for you with the `assert_no_structural_warnings` method.

Reply:
Okay I will see

Reply:
The assert_no_structural_warnings test is failing, there is a unit mismatch in the material accumulation term, where the DerivativeVar is being applied

Reply:
Good - now I know you actually did the test. To be sure however, you need to provide the traceback of the error message so we can be sure.

What I expect you are seeing is the result of an edge case in Pyomo, and unfortunately affects all DAE models with units. Basically, the issue arises in the numerical approximation of the derivative whee we get:

`der_var[t] == (v[t] - v[t-1])/([t] - [t-1])`

For this case, `der_var` has units of `mass/time` and `v` has units of `mass` (or `moles` as appropriate). Time should have units of `time`, which would make this consistent, but Pyomo `Sets` (and thus `ContinuousSets`) do not have units (semantically, as Set does not have units, although the elements inside it might). Unfortunately, from a coding sense, this is actually really hard to fix, so for now we have been living with the issue).

Reply:
`pyomo.util.check_units:check_units.py:251 Units problem with expression fs.solex.mscontactor.organic_material_accumulation[60.0,3,liquid,Dy] - (0.499999999999981*fs.solex.mscontactor.organic_material_holdup[48.0,3,liquid,Dy] - 0.8285814518273972*fs.solex.mscontactor.organic_material_holdup[48.477718,3,liquid,Dy] + 0.6396718464360766*fs.solex.mscontactor.organic_material_holdup[50.376161,3,liquid,Dy] - 0.6860614273515556*fs.solex.mscontactor.organic_material_holdup[53.255698,3,liquid,Dy] + 0.9698922731138939*fs.solex.mscontactor.organic_material_holdup[56.345571,3,liquid,Dy] - 2.1365879070378115*fs.solex.mscontactor.organic_material_holdup[58.817579,3,liquid,Dy] + 1.5416666666667163*fs.solex.mscontactor.organic_material_holdup[60.0,3,liquid,Dy])`

Reply:
unit assertion error

Reply:
I need the full output from the unit check to be certain. It should tell you what the unit inconsistency is too (i.e. units x are not compatible with units y). You need to use the Pyomo `assert_units_consistent` function for this. It would also help to know which constraint that is coming from.

Reply:
`InconsistentUnitsError: Error in units found in expression: fs.solex.mscontactor.aqueous_material_accumulation[0.477718,1,liquid,H2O] - (-1.3155449435149056*fs.solex.mscontactor.aqueous_material_holdup[0.0,1,liquid,H2O] + 1.0466419563576284*fs.solex.mscontactor.aqueous_material_holdup[0.477718,1,liquid,H2O] + 0.3396521657396914*fs.solex.mscontactor.aqueous_material_holdup[2.376161,1,liquid,H2O] - 0.10143365070535666*fs.solex.mscontactor.aqueous_material_holdup[5.255698,1,liquid,H2O] + 0.04718598891201106*fs.solex.mscontactor.aqueous_material_holdup[8.345571,1,liquid,H2O] - 0.025592017688069048*fs.solex.mscontactor.aqueous_material_holdup[10.817579,1,liquid,H2O] + 0.009090500899000946*fs.solex.mscontactor.aqueous_material_holdup[12.0,1,liquid,H2O]): mole / hour not compatible with mole.`

Reply:
OK, so then it is the expected error in Pyomo. At least that means the original concern is not an issue.

Reply:
So Andrew, what should I do next about this structural test fail? Because this will create a pytest fail everytime in git

Reply:
`assert_no_structural_warnings(ignore_unit_consistency=False)`

We have at least now confirmed that he only unit consistency issue if the expected one from the DAE tools ,so the best we can do now is to just ignore it.

Reply:
I think you meant `ignore_unit_consistency=True`